### PR TITLE
scheme.go: Wrap all errors with context

### PR DESCRIPTION
**Pull request by brancz**
@bwplotka @metalmatze @squat @kakkoyun

### CODE REVIEW

File: scheme.go [hunk 1]
```diff
@@ -303,14 +310,14 @@ func (rs *replicationScheme) ensureObjectReplicated(ctx context.Context, objectN
...
-		return err
+		return fmt.Errorf("get %v from origin bucket: %w", objectName, err)
 	}
 
 	defer r.Close()
 
 	err = rs.toBkt.Upload(ctx, objectName, r)
 	if err != nil {
-		return err
+		return fmt.Errorf("upload %v to target bucket: %w", objectName, err)
```
Comment (bwplotka) [inline scheme.go]: this all should be `errors.Errorf` I think :thinking:

Comment (brancz) [reply scheme.go]: I may be totally misunderstanding but this doesn’t sound like that to me https://blog.golang.org/go1.13-errors

Comment (bwplotka) [reply scheme.go]: wooow, this is more complex then I expected. Looks like `"github.com/pkg/errors"` and `errors` are literally incompatible now. Also just learn that if you use ` `"github.com/pkg/errors.Wrap"`` and then on top caller you log this and if you use `%s` it's different then if you use `%v`... for %v it is stack trace even 0.o 

So yea, you are right I guess - let's choose one and use it consistently...

Comment (bwplotka) [reply scheme.go]: :scream:  https://github.com/golang/go/issues/31778#issuecomment-488326644

Comment (bwplotka) [reply scheme.go]: Actually this should fix the compatibility: https://github.com/pkg/errors/pull/206

**Pull request merged**